### ISSUE TITLE
PnP Caching is not working on Search query

### ISSUE THREAD

**Issue by Dangerous-Mind** [pnp/pnpjs#2229]
### Category
- [ ] Enhancement
- [x] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [ 3.2.0 ]

Please specify what version(s) of SharePoint you are targeting: [ Online ]

### Expected / Desired Behavior / Question
```
spfi().using(SPFx(this.Context), Caching({
  store: "local",
  keyFactory: () => "SomeKey",
  expireFunc: () => dateAdd(new Date(), "minute", 5)
})).search(query)
```
Should be cached and create a key in the local storage.

### Observed Behavior
The search function is not cached in local or session.

From PnP, the code on cached.js we have a code that only works on GET requests
The search function creates a POST request.

```
// only cache get requested data
if (!/get/i.test(init.method)) {
    return [url, init, result];
}
```

**Comment by patrick-rodgers** [comment]
Please have a [look at the docs](https://pnp.github.io/pnpjs/sp/search/#search-result-caching) and see if this helps. Let us know if you have any issues with the described functionality.

**Comment by Dangerous-Mind** [comment]
> Please have a [look at the docs](https://pnp.github.io/pnpjs/sp/search/#search-result-caching) and see if this helps. Let us know if you have any issues with the described functionality.

Hi @patrick-rodgers 

The **searchWithCaching** is not available on V3. We were migrating from V2 were we used the searchWithCaching when we found this issue.

The documentation is kinda messi at this point. I've seen several things that are not correct in there.
For instance it says that to [save a multi user field](https://pnp.github.io/pnpjs/sp/items/#user-fields) we need to use "result:" and in the code it has a array of ID's without "result" has it's suppose to be on V3.

**Comment by juliemturner** [comment]
@Dangerous-Mind  -- We gladly accept PRs for documentation updates we missed...

**Comment by patrick-rodgers** [comment]
Well shoot - so we removed it from the library but not my brain. Ok, let me look into this then, sorry that you hit this issue.

**Comment by patrick-rodgers** [comment]
Got this fixed and will go out in 3.3. Looked back at the history and I apparently tested this before adding the /get/ check in the caching and never tested the search caching again. That's a miss on our part, but now fixed. With the addition of the X-PnP-CacheAlways header we can now cover any need to support caching for non-get requests in a way that is also open to consumers of the library.

**Comment by Dangerous-Mind** [comment]
Only managed to test this one today.
It's working fine now. 

Thanks @patrick-rodgers
